src/OrleansRuntime/LogViews/NotificationTracker.cs

```diff
@@ -0,0 +1,203 @@
+﻿using Orleans.LogViews;
+using Orleans.MultiCluster;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.LogViews
+{
+
+    /// <summary>
+    /// Helper class for tracking notifications that a log view grain sends to other clusters after updating the log.
+    /// </summary>
+    internal class NotificationTracker
+    {
+        private IProtocolServices services;
+        private Dictionary<string, NotificationWorker> sendworkers;
+        private int maxnotificationbatchsize;
+
+        public NotificationTracker(IProtocolServices services, MultiClusterConfiguration configuration, int maxnotificationbatchsize)
+        {
+            this.services = services;
+            sendworkers = new Dictionary<string, NotificationWorker>();
+            this.maxnotificationbatchsize = maxnotificationbatchsize;
+
+            foreach (var x in configuration.Clusters)
+                if (x != services.MyClusterId)
+                {
+                    services.Verbose("Now sending notifications to {0}", x);
+                    sendworkers.Add(x, new NotificationWorker(services, x, maxnotificationbatchsize));
+                }
+        }
+
+        public void BroadcastNotification(INotificationMessage msg, string exclude = null)
+        {
+            foreach (var kvp in sendworkers)
+                if (kvp.Key != exclude)
+                {
+                    var w = kvp.Value;
+                    w.Enqueue(msg);
+                }
+        }
+
+        /// <summary>
+        /// last observed exception, or null if last notification attempts were successful for all clusters
```

Comment:
OK. Rationale or pointers to somewhere feels like would be nice here. Especially for those who come later and if they need to know this when working on other storage backends.